packages/@aws-cdk/aws-pipes-sources-alpha/lib/kinesis.ts

```diff
@@ -0,0 +1,72 @@
+import { IPipe, SourceConfig } from '@aws-cdk/aws-pipes-alpha';
+import { IRole } from 'aws-cdk-lib/aws-iam';
+import { IStream } from 'aws-cdk-lib/aws-kinesis';
+import { KinesisStartingPosition } from './enums';
+import { StreamSource, StreamSourceParameters } from './streamSource';
+
+/**
+ * Parameters for the Kinesis source.
+ */
+export interface KinesisSourceParameters extends StreamSourceParameters {
+  /**
+   * (Streams only) The position in a stream from which to start reading.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-startingposition
+   */
+  readonly startingPosition: KinesisStartingPosition;
+
+  /**
+   * With StartingPosition set to AT_TIMESTAMP, the time from which to start reading, in ISO 8601 format.
+   *
+   * @example
+   * '2025-01-01T00:00:00Z'
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-startingpositiontimestamp
+   * @default - no starting position timestamp
+   */
+  readonly startingPositionTimestamp?: string;
```

Comment:
Do we not have any type for this? I guess there would be very little to add but it seems like a thing we might have in core.

Reply:
I updated this to `Date`.